benchmarks/src/bench/json_multi_export.rs

```diff
@@ -4,106 +4,117 @@ use std::{
     path::PathBuf,
 };
 
-use mongodb::{Client, Collection, Database};
+use anyhow::Result;
+use futures::stream::{FuturesUnordered, StreamExt, TryStreamExt};
+use mongodb::{bson::doc, Client, Collection, Database};
 
-use crate::{
-    bench::{parse_json_file_to_documents, Benchmark, COLL_NAME, DATABASE_NAME},
-    error::Result,
-};
+use crate::bench::{parse_json_file_to_documents, Benchmark, COLL_NAME, DATABASE_NAME};
 
 const TOTAL_FILES: usize = 100;
 
 pub struct JsonMultiExportBenchmark {
     db: Database,
     coll: Collection,
-    num_threads: usize,
 }
 
 // Specifies the options to a `JsonMultiExportBenchmark::setup` operation.
 pub struct Options {
-    pub num_threads: usize,
     pub path: PathBuf,
     pub uri: String,
 }
 
+#[async_trait::async_trait]
 impl Benchmark for JsonMultiExportBenchmark {
     type Options = Options;
 
-    fn setup(options: Self::Options) -> Result<Self> {
-        let client = Client::with_uri_str(&options.uri)?;
+    async fn setup(options: Self::Options) -> Result<Self> {
+        let client = Client::with_uri_str(&options.uri).await?;
         let db = client.database(&DATABASE_NAME);
-        db.drop()?;
+        db.drop(None).await?;
 
         let coll = db.collection(&COLL_NAME);
 
+        let mut tasks = FuturesUnordered::new();
+
         for i in 0..TOTAL_FILES {
-            let json_file_name = options.path.join(format!("ldjson{:03}.txt", i));
-            let file = File::open(&json_file_name)?;
+            let path = options.path.clone();
```

Comment:
Importing all this data takes an _extremely_ long time, so I expedited it by using the powers of async to parallelize it. It's somewhat ironic that the setup for the benchmark was so slow that we actually had to do the optimization that we do in the benchmark itself to save time!

Reply:
Nope; the old version was incredibly slow too from testing it, but we didn't have such low hanging fruit to speed it up like we do now